vsutil/info.py

```diff
@@ -60,6 +60,21 @@ def get_plane_size(frame: Union[vs.VideoFrame, vs.VideoNode], /, planeno: int) -
     return width, height
 
 
+@func.disallow_variable_format
+def get_sample_type(clip: vs.VideoNode, /) -> vs.SampleType:
+    """Returns the sample type of a VideoNode as an SampleType.
+
+    >>> src = vs.core.std.BlankClip(format=vs.YUV420P16)
+    >>> get_sample_type(src)
+    <SampleType.INTEGER: 0>
+
+    :param clip:  Input clip.
+
+    :return:      Sample type of the input `clip`.
+    """
+    return clip.format.sample_type
```

Comment:
I have to assume the sole purpose for this function is to make use of the `disallow_variable_format` decorator? Because I don't see the advantage in `get_sample_type(clip)` over `clip.format.sample_type`.

Reply:
`get_depth` uses the same logic though, so I didn't think it was necessary.

Reply:
`get_depth` is a very common operation, so saving some typing there seemed like a good idea at the time, with the additional benefit of catching breaking changes in the Vapoursynth API.
I don’t think `get_sample_type` would see the same kind of usage tbh.

Reply:
> `get_depth` uses the same logic though, so I didn't think it was necessary.

Well, I'm with you on that being borderline, but the main reason for why I consider `get_depth` to be warranted is that it uses a different name for this info as opposed to `clip.format.bits_per_sample`. Now, I personally would be fine with the latter, but I at least see some use of having that alias.